### PR TITLE
Update README for updated library usage.

### PR DESCRIPTION
**Pull request by pirgeo**

### CODE REVIEW

**Review by arminru — APPROVED** [review]

**Review by arminru — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -15,64 +15,123 @@ It shows how to create metrics lines that can be sent to a [Dynatrace metrics in
...
+#### Metric line creation
+
+1. `create`: Instantiates a `MetricLineBuilder` with an optional `MetricLinePreConfiguration`.
+2. `metricKey`: Sets and normalizes the metric key of the metric line.
+3. `dimensions` / `dimension`:  Sets one or more dimensions specific to this metric.
+   * When using a metric line pre-configuration: Default dimensions and Dynatrace metadata will be merged with the newly set dimension(s). See [the section on dimension precedence](#dimension-precedence) below.
+   * When setting dimensions with the same dimension key multiple times, the dimension value that was set last is used for that dimension key.
+4. `count` / `gauge`: Sets the type of the metric line.
+  Use `gauge()` for gauges and summary statistics, and `count()` for counters.
+5. Sets the value:
```
[the remark this answers is not in the document]
Comment (pirgeo) [reply README.md:90]: Ah, I tried to keep it the same as above: `Instantiates`, `Sets`, `Serializes`, etc. I think we can change all of them or none of them :D

File: README.md [hunk 2]
```diff
@@ -15,64 +15,123 @@ It shows how to create metrics lines that can be sent to a [Dynatrace metrics in
...
+The following settings can be set upon `MetricLinePreConfiguration` construction:  
 
-* `withPrefix`: A prefix that is prepended to each metric key (separated by a `.`).
-* `withDefaultDimensions`: These dimensions will be added to all metrics created by the factory (see [the section on dimension precedence](#dimension-precedence) below).
-* `withDynatraceMetadata`: With this setting enabled, the library will connect to the Dynatrace OneAgent, if installed,
-  and retrieve process and host identifiers that are added as dimensions on all metrics to correlate them accordingly.
-  More information on the underlying feature that is used by the library can be found in the
+* `prefix`: A prefix that is prepended to each metric key (separated by a `.`).
+* `defaultDimensions`: Default dimensions that will be added to every metric using this `MetricLinePreConfiguration` object (see [the section on dimension precedence](#dimension-precedence) below).
+* `dynatraceMetadataDimensions`: With this setting enabled, the library will connect to the Dynatrace OneAgent, if available, and retrieve process and host identifiers.
```
Comment (pichlermarc) [inline README.md:30]:
```suggestion
* `dynatraceMetadataDimensions`: With this setting enabled, the library will connect to the Dynatrace OneAgent, if available, and retrieve process and host identifiers.
```

File: README.md [hunk 3]
```diff
@@ -15,64 +15,123 @@ It shows how to create metrics lines that can be sent to a [Dynatrace metrics in
...
+    MetricLinePreConfiguration.builder()
+        .defaultDimensions(defaultDims)
+        .dynatraceMetadataDimensions()
+        .prefix("prefix")
         .build();
 ```
 
-This factory can then be used to create new `Metric.Builder` objects, which represent one data point before serialization.
-To create metric lines from data points, use a pattern like the following:
+This pre-configuration can then be used to create `MetricLineBuilder` instance, which represent one data point before serialization.
```
Comment (pichlermarc) [inline README.md:48]:
```suggestion
This pre-configuration can then be used to create `MetricLineBuilder` instances, which represent one data point before serialization.
```

**Review by pichlermarc — APPROVED** [review]
I also tried out the examples; looks great! :slightly_smiling_face:

**Comment by pichlermarc** [comment]
(whoops, looks like I was too late :sweat_smile:)

**Pull request merged**